core/psh/psh.c

```diff
@@ -102,16 +102,16 @@ int main(int argc, char **argv)
 
 		/* Run app */
 		if ((app = psh_findapp(base)) != NULL) {
-			app->run(argc, argv);
+			err = app->run(argc, argv);
 		}
 		else {
 			fprintf(stderr, "psh: %s: unknown command\n", argv[0]);
 			break;
 		}
 
-	} while (tcpid == -1 && strcmp(base, "pshlogin") == 0);
+	} while (psh_common.tcpid == -1 && strcmp(base, "pshlogin") == 0);
```

Comment:
Maybe we would like to cache result of `strcmp(base, "pshlogin")` in some integer on the beginning and use it here instead? We would save some CPU cycles this way.

Reply:
Cached it as a flag in the beginning. Thanks!